rclcpp/include/rclcpp/copy_all_parameters.hpp

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 Open Navigation LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__COPY_ALL_PARAMETERS_HPP_
+#define RCLCPP__COPY_ALL_PARAMETERS_HPP_
+
+#include <vector>
+#include <string>
+#include "rcl_interfaces/srv/list_parameters.hpp"
+#include "rclcpp/parameter.hpp"
+
+namespace rclcpp
+{
+
+/**
+ * @brief A method to copy all parameters from one node (parent) to another (child).
+ * May throw parameter exceptions in error conditions
```

Comment:
It would be good to enumerate the errors, or probably better to just mention which other methods one could look at to enumerate them, e.g. list/get/set parameters.

Reply:
I actually just went through and found that it cannot throw. Since the nodes were already accepted by one node, this logic cannot throw since its already validated to exist via a valid name and parameter type. The only way it could throw is if a param is attempted to declare if it already exists, if we were to change some of this logic from review comments below. We can follow up down there, but as long as we check if it has a param before declaring it, we're all good